src/kernel/outwards.jl

```diff
@@ -32,21 +32,90 @@ is occupied.
 
 Pass grid name `Symbol`s to `R` and `W` type parameters to use specific grids.
 """
-struct OutwardsDispersal{R,W,S<:Stencils.AbstractKernelStencil} <: SetNeighborhoodRule{R,W}
+struct _Mask end
+const NoMask = nothing
+
+struct OutwardsDispersal{R,W,S<:Stencils.AbstractKernelStencil, M} <: SetNeighborhoodRule{R,W}
     stencil::S
+    mask_flag::M
+end
+
+# Constructors for OutwardsDispersal
+function OutwardsDispersal{R,W}(stencil::S; mask_flag=NoMask) where {R,W,S<:Stencils.AbstractKernelStencil}
+    OutwardsDispersal{R,W,S,typeof(mask_flag)}(stencil, mask_flag)
 end
-function OutwardsDispersal{R,W}(; kw...) where {R,W}
-    OutwardsDispersal{R,W}(DispersalKernel(; kw...))
+
+function OutwardsDispersal{R,W}(; mask_flag=NoMask, kw...) where {R,W}
+    stencil = DispersalKernel(; kw...)
+    OutwardsDispersal{R,W,typeof(stencil),typeof(mask_flag)}(stencil, mask_flag)
 end
 
 @inline function applyrule!(data, rule::OutwardsDispersal{R,W}, N, I) where {R,W}
     N == zero(N) && return nothing
+    
+    # Check if the current cell is masked, skip if it is
+    mask_data = if rule.mask_flag === NoMask nothing else mask(data) end
+    if !isnothing(mask_data) && !mask_data[I...]
+        return nothing
+    end
+    
     sum = zero(N)
     for (offset, k) in zip(offsets(rule), kernel(rule))
-        @inbounds propagules = N * k
-        @inbounds add!(data[W], propagules, I .+ offset...)
-        sum += propagules
+        target = I .+ offset
+        (target_mod, inbounds) = inbounds(data, target)
+        if inbounds && (isnothing(mask_data) || mask_data[target_mod...])
+            @inbounds propagules = N * k  
+            @inbounds add!(data[W], propagules, target_mod...)  
+            sum += propagules
+        end
     end
     @inbounds sub!(data[W], sum, I...)
     return nothing
 end
+
+# @inline function applyrule(data, rule::OutwardsDispersal{R,W}, N, I) where {R,W}
+#     applyrule!(data, rule, N, I)
+# end
+
+################# TESTING #################
+# Define a mask
+mask_data = fill(true, 10, 10)
+for i in 3:9 
+    mask_data[2, i] = false
+    mask_data[3, i] = false
+    mask_data[4, i] = false
+    mask_data[5, i] = false
+    mask_data[6, i] = false
+    mask_data[7, i] = false
+    mask_data[8, i] = false
+end
+
+# Create OutwardsDispersal with and without a mask
+outdisp_with_mask = OutwardsDispersal(
+    formulation=ExponentialKernel(λ=0.0125),
+    distancemethod=AreaToArea(30),
+    mask_flag=_Mask()
```

Comment:
If you want to use it directly like this I would just make it `Mask`, underscore is if its internal